### PR TITLE
add clear cache to user dev

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes RNBW-2906

context: https://rainbowhaus.slack.com/archives/C02C2FVC6N6/p1648140026260999

## What changed (plus any additional context for devs)
giving users access to clear local cache via dev settings

## PoW (screenshots / screen recordings)
https://cloud.skylarbarrera.com/Screen-Shot-2022-03-24-12-12-29.67.png

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

**Comment by osdnk** [comment]
Why do we need this?

**Comment by osdnk** [comment]
Ok, I got the motivation. Please, include this in the ticket or PR description.

**Review by pugson — APPROVED** [review]
PR has a link to the thread on slack so lgtm 🌈

File: src/components/settings-menu/UserDevSection.js [hunk 1]
```diff
@@ -60,6 +61,7 @@ const UserDevSection = props => {
         </Column>
       </ListItem>
       {testnetsEnabled && <NetworkSection {...props} />}
+      <ListItem label="💥 Clear local storage" onPress={AsyncStorage.clear} />
```
Comment (pugson) [inline src/components/settings-menu/UserDevSection.js:64]: wait hold up...
```suggestion
      <ListItem label="💥 Clear local storage" onPress={AsyncStorage.clear()} />
```

**Review by pugson — CHANGES_REQUESTED** [review]

File: src/components/settings-menu/UserDevSection.js (hunk 1 above)
Comment (osdnk) [reply src/components/settings-menu/UserDevSection.js:64]: no

Comment (pugson) [reply src/components/settings-menu/UserDevSection.js:64]: is this not the correct way to call this function? https://react-native-async-storage.github.io/async-storage/docs/api/#clear

Comment (pugson) [reply src/components/settings-menu/UserDevSection.js:64]: or is `onPress={AsyncStorage.clear}` the same as?
```suggestion
      <ListItem label="💥 Clear local storage" onPress={() => AsyncStorage.clear()} />
```

Comment (osdnk) [reply src/components/settings-menu/UserDevSection.js:64]: https://stackoverflow.com/questions/40275154/react-native-listitem-onpress-why-is-this-function-executing

Comment (pugson) [reply src/components/settings-menu/UserDevSection.js:64]: yes the first suggestion was wrong. haven't seen this syntax without using an arrow function before. TIL

**Review by pugson — APPROVED** [review]

**Review by tchayen — APPROVED** [review]
Looks good. As Michał mentioned – I'd add MMKV cleaning as well (or maybe combine them even here? For devs it makes sense to sometimes have them separated, but if we are exposing it to the user side then I don't see any reasons).

**Review by osdnk — APPROVED** [review]

**Pull request merged**